File: c++/1619meanOfArrayAfterRemoving.cc
#include <iostream>
#include <sstream>
#include <vector>
#include <string>
#include <queue>
#include <climits>
#include <stack>
#include <algorithm>
#include <cmath>
#include <set>
#include <unordered_map>
#include <list>
#include <unordered_set>
#include <map>
#include <functional>
#include <bitset>
#include <numeric>
#include <deque>
#include <mutex>
#include <utility>
#include <memory>
#include <cstring>

using namespace std;

static int x = [] () {ios::sync_with_stdio(false); cin.tie(0); return 0;} ();
class Solution {
 public:
  double trimMean(vector<int>& arr) {
    int n = arr.size(), r = n * 0.05;
    sort(arr.begin(), arr.end());
    double sum = .0;
    for (int i = r, stop = n - r; i < stop; ++i) sum += arr[i];
    return sum / (n - (r << 1));
  }
};

int main() {
  Solution s;
  
  return 0;
}

File: python3/1619meanOfArrayAfterRemoving.py
class Solution:
    def trimMean(self, arr: List[int]) -> float:
      n = len(arr)
      r = int(n / 20)
      s = 0
      list.sort(arr)
      for i in range(r, n - r):
        s += arr[i]
      return s / (n - r * 2)


# more "python" code from leetcode
class Solution:
    def trimMean(self, arr: List[int]) -> float:
        l = len(arr)
        toremove = ((l * 5) // 100)
        remaining = sorted(arr)[toremove:l-toremove]
        s = 0
        for n in remaining:
            s += n
        return s / (l-toremove*2)
